### PR TITLE
Add support for Groovy based migration scripts

### LINKED ISSUE #26: Add support for groovy based migration script
The goal of this issue is to extend the migtool adding the capability to execute a groovy script to carry out a migration task. 

Main goal are the following: 

* the migration script follows the same file name rules as plain SQL-based migration scripts, but uses `.groovy` extension instead `sql` 
* the identical checksum and validation mechanism is applied  
* once a groovy migration entry needs to be executed a [Groovy Sql instance is created](https://docs.groovy-lang.org/latest/html/documentation/sql-userguide.html#_connecting_to_the_database).
* the groovy script is parsed and executed using a [GroovyShell](https://docs.groovy-lang.org/latest/html/documentation/guide-integrating.html#integ-groovyshell). The Groovy *Sql* instance is bound the script execution context as `sql`, so that it can be used to carry out the migration script

### PR DESCRIPTION
**Pull request by tcrespog**
Closes #26.

### CODE REVIEW

File: src/main/java/io/seqera/migtool/MigRecord.java [hunk 1]
```diff
@@ -15,10 +15,23 @@
  */
 class MigRecord implements Comparable<MigRecord> {
 
+    enum Language {
+        sql,
+        groovy;
```
Comment (pditommaso) [inline src/main/java/io/seqera/migtool/MigRecord.java:20]: Let's use uppercase that's more idiomatic, here we don't need to care about ser-deser of those names

**Review by pditommaso — CHANGES_REQUESTED** [review]
Looks neat 👍 Just a minor comment

File: src/main/java/io/seqera/migtool/MigRecord.java (hunk 1 above)
Comment (tcrespog) [reply src/main/java/io/seqera/migtool/MigRecord.java]: :+1:. Done.

**Comment by tcrespog** [comment]
Now that I see tests having to do with native compilation [failing](https://github.com/seqeralabs/migtool/actions/runs/3066188637/jobs/4951082610)... I think Groovy support is incompatible with native compilation, especially to the extent that we want to use it (run any scripts) :disappointed:.

**Comment by pditommaso** [comment]
The guru here is @jorgeaguileraseqera!

**Comment by jorgeaguileraseqera** [comment]
I don't think Graalvm will allow GroovyShell execute arbitrary code

With some little effort, I was able to run the groovy code in a native way but when the GroovyShell parses the resource it fires an Exception:
> No classes have been predefined during the image build to load from bytecodes at runtime

**Comment by tcrespog** [comment]
Thanks for looking into it, Jorge.

> With some little effort, I was able to run the groovy code in a native way but when the GroovyShell parses the resource it fires an Exception

What a pity! However, I think it's acceptable so long as we are aware of the limitation. As far as SQL scripts run fine using the native image, we could live with that.

**Comment by pditommaso** [comment]
Check the paragraph "Mixing AOT and JIT"

https://medium.com/graalvm/java-on-truffle-going-fully-metacircular-215531e3f840

**Pull request merged**